src/lint/collect-header-diagnostics.ts

```diff
@@ -0,0 +1,104 @@
+import type { LintingError } from './algorithm-error-reporter-type';
+
+import { getLocation } from './utils';
+
+const ruleId = 'header-format';
+
+export function collectHeaderDiagnostics(
+  dom: any,
+  headers: { element: Element; contents: string }[]
+) {
+  let lintingErrors: LintingError[] = [];
+
+  for (let { element, contents } of headers) {
+    if (!/\(.*\)$/.test(contents) || / Operator \( `[^`]+` \)$/.test(contents)) {
+      continue;
+    }
+
+    function indexToLineAndColumn(index: number) {
```

Comment:
I'd prefer this be a pure function of the text content, the index into the text content, and the starting { line, column, offset } of the text content.

Reply:
That splits the logic in an awkward way. I've changed it to be pure and take an element location, though.